core/src/main/java/org/elasticsearch/search/internal/InternalSearchResponse.java

```diff
@@ -19,148 +19,50 @@
 
 package org.elasticsearch.search.internal;
 
+import org.elasticsearch.action.search.SearchResponseSections;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.xcontent.ToXContent;
-import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.search.SearchHits;
-import org.elasticsearch.search.aggregations.Aggregations;
 import org.elasticsearch.search.aggregations.InternalAggregations;
-import org.elasticsearch.search.profile.ProfileShardResult;
 import org.elasticsearch.search.profile.SearchProfileShardResults;
 import org.elasticsearch.search.suggest.Suggest;
 
 import java.io.IOException;
-import java.util.Collections;
-import java.util.Map;
 
-public class InternalSearchResponse implements Streamable, ToXContent {
+/**
+ * {@link SearchResponseSections} subclass that can be serialized over the wire.
+ */
+public class InternalSearchResponse extends SearchResponseSections implements Writeable, ToXContent {
 
     public static InternalSearchResponse empty() {
         return new InternalSearchResponse(SearchHits.empty(), null, null, null, false, null, 1);
     }
 
-    private SearchHits hits;
-
-    private InternalAggregations aggregations;
-
-    private Suggest suggest;
-
-    private SearchProfileShardResults profileResults;
-
-    private boolean timedOut;
-
-    private Boolean terminatedEarly = null;
-
-    private int numReducePhases = 1;
-
-    private InternalSearchResponse() {
-    }
-
     public InternalSearchResponse(SearchHits hits, InternalAggregations aggregations, Suggest suggest,
```

Comment:
Maybe, I tried to limit the unrelated changes here as it will go in our branch. Does the need for Nullable come from my changes?